cpp/src/cylon/mapreduce/mapreduce.cpp

```diff
@@ -0,0 +1,1033 @@
+//
+// Created by niranda on 11/22/21.
+//
+
+#include <cylon/thridparty/flat_hash_map/bytell_hash_map.hpp>
+#include "cylon/mapreduce/mapreduce.hpp"
+#include "cylon/util/macros.hpp"
+
+#include <arrow/buffer_builder.h>
+#include <arrow/visitor_inline.h>
+
+namespace cylon {
+namespace mapred {
+
+template<typename ArrowT, typename Visitor>
+void CombineVisit(const std::shared_ptr<arrow::Array> &value_col, const int64_t *local_group_ids,
+                  Visitor &&visitor) {
```

Comment:
Just asking out of curiosity... 

Would like to know whether the compiler inlines the visitor at 23. 
Would making visitor const give any compilation time advantage?

Reply:
yes, from what I've seen previously, the compiler does inline that visitor call.